singleton-service/src/service/report.rs

```diff
@@ -64,10 +64,26 @@ pub fn build_report_request(report: &Report) -> Result<Request, anyhow::Error> {
     let creds = Credentials::ServiceToken(ServiceToken::from(report.service_token()));
     let svc = Service::new(report.service_id(), creds);
     let mut app_usage = vec![];
-    for (user_key, usage) in report.usages().iter() {
-        let application = Application::from(UserKey::from(user_key.as_str()));
+    for (app_id, usage) in report.usages().iter() {
+        let app;
+        match app_id {
+            AppIdentifier::UserKey(_user_key) => {
+                debug!("AppIdentifier UserKey: {}", app_id.as_string());
+                app = Application::from_user_key(app_id.as_string())
+            }
+            AppIdentifier::AppId(_app_id, None) => {
+                debug!("AppIdentifier AppId : {}", app_id.as_string());
+                app = Application::from_app_id(app_id.as_string())
+            }
+            AppIdentifier::AppId(_app_id, _app_key) => {
+                let app_key = app_id.as_string();
+                debug!("AppIdentifier AppId+AppKey : {}", app_key);
+                let keys = app_key.split('_').collect::<Vec<_>>();
+                app = Application::from_app_id_and_key(keys[0], keys[1])
+            }
+        }
```

Comment:
Duplicate code with the same concerns mentioned above.

Reply:
fixed